drivers/i2c/i2c_ite_it8xxx2.c

```diff
@@ -735,6 +735,22 @@ bool __soc_ram_code fifo_mode_allowed(const struct device *dev,
 }
 #endif
 
+#ifdef CONFIG_I2C_SHELL
+void __soc_ram_code i2c_scan_transaction_detect(const struct device *dev,
```

Comment:
The calling routine `i2c_it8xxx2_transfer` is not linked into the `__ram_code` section.  Does this function need to be linked into ILM?